DopplerCustomDomain.Test/CustomDomainEndpointTest.cs

```diff
@@ -393,5 +418,113 @@ public async Task GET_domain_ip_resolution_should_return_BadRequest_when_it_reso
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var domainName = fixture.Create<string>();
+            var expectedRuleType = RuleType.HttpsOnly;
+            var ruleType = expectedRuleType.ToString();
+            var expectedService = "relay-actions-api_service_prod@docker";
+            var service = "relay-tracking";
+
+            var customDomainProviderServiceMock = CreateCustomDomainProviderServiceMock();
+            var dnsResolutionValidatorMock = CreateDnsResolutionValidatorMock();
+            dnsResolutionValidatorMock.Setup(x => x.IsNamePointingToOurServiceAsync(domainName)).ReturnsAsync(true);
+
+            using var appFactory = _factory.WithBypassAuthorization();
+
+            var client = CreateHttpClient(
+                appFactory,
+                customDomainProviderService: customDomainProviderServiceMock.Object,
+                dnsResolutionValidator: dnsResolutionValidatorMock.Object);
+
+            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost/{domainName}");
+            request.Content = JsonContent.Create(new { service, ruleType });
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(domainName, expectedService, expectedRuleType), Times.Once);
+            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<RuleType>()), Times.Once);
+        }
+
+        [Fact]
+        public async Task PUT_domain_should_return_BadRequest_and_not_store_domain_when_it_does_not_resolve_to_our_IP()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var domainName = fixture.Create<string>();
+            var ruleType = "HttpsOnly";
+            var service = "relay-tracking";
+
+            var customDomainProviderServiceMock = CreateCustomDomainProviderServiceMock();
+            var dnsResolutionValidatorMock = CreateDnsResolutionValidatorMock();
+            dnsResolutionValidatorMock.Setup(x => x.IsNamePointingToOurServiceAsync(domainName)).ReturnsAsync(false);
+
+            using var appFactory = _factory.WithBypassAuthorization();
+
+            var client = CreateHttpClient(
+                appFactory,
+                customDomainProviderService: customDomainProviderServiceMock.Object,
+                dnsResolutionValidator: dnsResolutionValidatorMock.Object);
+
+            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost/{domainName}");
+            request.Content = JsonContent.Create(new { service, ruleType });
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
```

Comment:
This assert is redundant, when the value is `null`, the next assert will fail.
See [avoid multiple asserts](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-asserts)

This comment applies to other tests too.

```suggestion
```

Reply:
Unnecessary assertion removed in #109

> This assert is redundant, when the value is null, the next assert will fail.

In this case, I think that this assertion is redundant because it should never be null. 

But if `response = null` would be a possibility, "the next assert will fail" is not a good reason because the next assert would fail with a cryptic "null pointer exception".